test/runner_mode.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+set -e
+
+echo "=== Testing runner mode functionality ==="
+
+# Test directory
+TEST_DIR=$(mktemp -d)
+trap 'rm -rf -- "$TEST_DIR"' EXIT
+
+echo "=== Test 1: Generate runner script for binst ==="
+./binst gen --config .config/binstaller.yml --type runner -o "$TEST_DIR/run-binst.sh"
+
+# Verify runner script was generated
+if [ ! -f "$TEST_DIR/run-binst.sh" ]; then
+    echo "ERROR: Runner script was not generated"
+    exit 1
+fi
+
+# Verify it's a runner script (not installer)
+if ! grep -q "download and run" "$TEST_DIR/run-binst.sh"; then
+    echo "ERROR: Generated script should contain 'download and run' in usage"
+    exit 1
+fi
+
+# Verify it has runner-specific usage patterns
+if ! grep -q -- "-- \[binary arguments\]" "$TEST_DIR/run-binst.sh"; then
+    echo "ERROR: Generated script should show binary arguments pattern"
+    exit 1
+fi
+
+# Verify it has example usage with --
+if ! grep -q "Example.*-- --help" "$TEST_DIR/run-binst.sh"; then
+    echo "ERROR: Generated script should have example with -- --help"
+    exit 1
+fi
+
+# Verify it doesn't have installer-specific options like -b (bindir)
+if grep -q "\-b.*bindir" "$TEST_DIR/run-binst.sh"; then
+    echo "ERROR: Runner script should not have -b bindir option"
+    exit 1
+fi
```

Comment:
Let's simplify this E2E test. 
Remove these checks. This kind of checks are tested by Go test.